larsimrad/BxDecay0/BaseRadioGen.h

```diff
@@ -281,24 +421,33 @@ namespace evgen {
         if (!node) continue;
         if (node->IsOverlapping()) continue;
 
-        std::string material_name = node->GetMedium()->GetMaterial()->GetName();
-        std::string volume_name = node->GetVolume()->GetName();
-        bool flag = std::regex_match(material_name, m_regex_material) && std::regex_match(volume_name, m_regex_volume);
-        if (!flag) continue;
+        auto good_mat = std::find(m_good_materials.begin(), m_good_materials.end(), node->GetMedium()->GetMaterial());
+        auto good_vol = std::find(m_good_volumes  .begin(), m_good_volumes  .end(), node->GetVolume());
+        bool good = good_mat != m_good_materials.end() and good_vol != m_good_volumes.end();
+
+        if (!good) continue;
```

Comment:
Suggest using the `cet::search_all` utility available via `#include "cetlib/container_algorithms.h"`.

```suggestion
        bool good = cet::search_all(m_good_materials, node->GetMedium()->GetMaterial()) and
                    cet::search_all(m_good_values, node->GetVolume());
        if (!good) continue;
```

Reply:
replaced with a new fcn